src/cocotb/handle.py

```diff
@@ -212,7 +212,7 @@ def _items(self) -> Iterable[Tuple[KeyType, SimHandleBase]]:
         self._discover_all()
         return self._sub_handles.items()
 
-    @lru_cache(maxsize=None)
```

Comment:
lru_cache keys on repr(obj), whereas cached_property attaches it to the actual object. This probably shouldn't come up in a real testbench though since everything goes through SimHandle and therefore the handle cache. But were this to change and we allowed deletion of the python handles, this would cause some issues like it did for this test

Reply:
The difference is ones a callable and the other is a property... you have a lot of code to update.

Reply:
I put a small workaround in by returning a lambda. Can I revert to how it was a long time ago with having a self.is_discovered variable? I don't think it's great to have properties return nothing